### PR TITLE
Reduce number of SQL queries

### PR DESCRIPTION
**Pull request by pitpalme**
Helpdesk start page creates many SQL queries to render facility links.
By optimizing QuerySet fetching open shifts this information is fetched
in one query, which reduces rendering time.

### CODE REVIEW

**Comment by pitpalme** [comment]
@christophmeissner Could you please check, if I've done nonsense? Using dummy data (`manage.py create_dummy_data 5`) it reduced number of SQL queries on helpdesk start page by almost 50% and query time by 1/3 to 1/2.

**Comment by asymmetric** [comment]
Seems like you forgot to add some files/commits...

**Comment by pitpalme** [comment]
@asymmetric Did I?

**Comment by pitpalme** [comment]
There's now two optimization attempts:

- prefetching organizations with shifts
- prefetching news for all organizations shown and filtering it in the app, instead of multiple SQL queries, one for each organization

In total changes cut response time to roughly one half, maybe even a little less (using ```ab -n 100 -C sessionid=... .../helpdesk/```).

Performance test done in docker containers showed unoptimized version also took significantly more CPU. Still low percentage, but optimized only about one third.

Review and merge (or criticism) would be nice 😄

**Comment by lwm** [comment]
I normally debug these / write tests for these optimizations with `CaptureQueriesContext`:

```
    with CaptureQueriesContext(connection) as capture:
        # use the ORM
    # have access to a count of: `capture.captured_queries`
```

My two cents ...

File: news/managers.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# coding=utf-8
```
Comment (christophmeissner) [inline news/managers.py:1]: Remove encoding comment, please!

```suggestion

```

File: organizations/views.py [hunk 2]
```diff
@@ -55,8 +55,9 @@ class FacilityView(DetailView):
     def get_context_data(self, **kwargs):
         context = super(FacilityView, self).get_context_data(**kwargs)
         shifts = Shift.open_shifts.filter(facility=self.object)
+        news = NewsEntry.objects.for_facilities([self.object])
         context['object'] = self.object
-        context['facility'] = get_facility_details(self.object, shifts)
+        context['facility'] = get_facility_details(self.object, shifts, news)  # FIXME
```
Comment (christophmeissner) [inline organizations/views.py:60]: FIXME???

File: scheduler/managers.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # coding: utf-8
```
Comment (christophmeissner) [inline scheduler/managers.py:1]: Remove
```suggestion

```

**Review by christophmeissner — CHANGES_REQUESTED** [review]
Just some litte change requests.

**Review by christophmeissner — APPROVED** [review]

**Pull request closed without merging**